uint8 mode_run 
uint8 manual_mode=1
uint8 auto_mode=2
